### PR TITLE
Support Devour Spell in Necromancer Healing

### PR DESCRIPTION
**Pull request by jandersson**
We give priority to devour if both consume flesh and devour are listed

### CODE REVIEW

**Comment by jandersson** [comment]
I also added some messages to `do_necromancer_ritual` which caused it to time out.

**Comment by jandersson** [comment]
And I removed the @necro attribute, since we can detect a persons guild with infomon

File: combat-trainer.lic [hunk 1]
```diff
@@ -705,8 +702,12 @@ class SpellProcess
...
     @custom_cast = "cast my #{game_state.weapon_name}" if game_state.casting_ignite
     @custom_cast = "cast my #{game_state.weapon_name}" if game_state.casting_rue
-    @custom_cast = "cast #{game_state.wounds[game_state.wounds.keys.max].first}" if game_state.casting_consume
-    cast(@custom_cast, @before, @after)
+    if game_state.casting_consume
+      if @necromancer_healing.key?('Consume Flesh') && ! @necromancer_healing.key?('Devour')
+        @custom_cast = "cast #{game_state.wounds[game_state.wounds.keys.max].first}"
+      end
+    end
+    cast(@custom_cast)
```
Comment (rpherbig) [inline combat-trainer.lic]: This looks like a merge error.

File: combat-trainer.lic [hunk 2]
```diff
@@ -818,11 +819,15 @@ class SpellProcess
   end
 
   def check_consume(game_state)
-    return unless @is_necro
+    return unless DRStats.necromancer? && @necromancer_healing
```
Comment (rpherbig) [inline combat-trainer.lic]: I'd prefer each clause to be a separate return, for readability

**Pull request merged**